src/index.ts

```diff
@@ -252,7 +252,9 @@ export class RedisRateLimiter extends RateLimiter {
       batch.exec((err, result) => {
         if (err) return reject(err);
 
-        const zRangeOutput = (addNewTimestamp ? result[2] : result[1]) as Array<unknown>;
+        const zRangeOutput = (addNewTimestamps > 0
+          ? result[2]
+          : result[1]) as Array<unknown>;
```

Comment:
I think this has a bug. The purpose of this ternary is to account for the extra `batch.zadd` call on line 246. The commands to Redis used to be as follows:
1. `zremrangebyscore` to clear old entries
2. `zadd` to add a new entry, _if we are adding a new timestamp_
3. `zrange` to get all current entries
4. `expire` to set a TTL

So the output of `zrange` is either at index 1 or 2 depending on whether we called `zadd` or not.

In your new implementation, `zadd` is called `addNewTimestamps` number of times. So I think we need to update this line to be:
```ts
const zRangeOutput = result[1 + addNewTimestamps] as Array<unknown>
```

Does that sound right?

I was also surprise that the tests did not catch this issue. I wonder if you could update your unit test such that it fails with the current code, but succeeds after updating?

Reply:
Seems correct, not sure about the tests.